modules/component/src/componentNext.ts

```diff
@@ -69,7 +33,17 @@ type Comparator = <S, P>(s1: S, p1: P, s2?: S, p2?: P) => boolean
 const defaultComparator: Comparator = () => true
 
 const arg2 = <A, B>(a: A, b: B) => b
+
+/**
+ * Type safe, composable, and view library agnostic component API
+ */
 export class ComponentNext<P1 extends ComponentProps> {
+  /**
+   * @param _init Function which returns initial state of the component
+   * @param _update Function which takes action and state, and returns new state
+   * @param _command Function which takes action and state, and returns new action for causing side effect
```

Comment:
the returned action is not necessarily for causing side-effect no? It's just another action